### PR TITLE
Store following certificate attributes: issuer, hash algorithm and hash of subject & public key

### PR DESCRIPTION
**Pull request by FdLSifu**
Pull-request associated to this issue: https://github.com/OpenSC/libp11/issues/251

### CODE REVIEW

File: src/p11_cert.c [hunk 1]
```diff
@@ -295,6 +299,33 @@ int pkcs11_store_certificate(PKCS11_TOKEN *token, X509 *x509, char *label,
...
+        ckm_md = CKM_SHA256;
+    case NID_sha512:
+        ckm_md = CKM_SHA512;
+    case NID_sha384:
+        ckm_md = CKM_SHA384;
+    default:
+        ckm_md = 0;
+    }
+
+    /* Set hash algorithm; default is SHA-1 */
```
Comment (frankmorgner) [inline src/p11_cert.c]: if SHA-1 is the default, then please make that explicit in the switch statement above, i.e. move `default:` above `case NID_sha1:`.

Comment (FdLSifu) [reply src/p11_cert.c]: Actually SHA-1 is the default CKA_NAME_HASH_ALGORITHM.
Meaning that if ckm_md is equal to 0, SHA-1 will be use. This comes from PKCS11.

**Comment by FdLSifu** [comment]
Done as requested

**Comment by frankmorgner** [comment]
Your switch statement misses the `break;`. Have you ever tested this?

**Comment by FdLSifu** [comment]
I have tested this by using function pkcs11_md2ckm from p11_pkey.c.
I have then copy back the code and you are right, break are missing.

**Comment by mtrojnar** [comment]
@frankmorgner Are you satisfied with the implementation of changes you requested?

**Comment by frankmorgner** [comment]
The code itself looks good, but I did not check the functionality with the PKCS#11 spec.

**Pull request merged**